docs/design/api-versions.md

```diff
@@ -32,8 +40,169 @@ For each API spec, generated `AssignPropertiesTo*()` method  will read the value
+``` go
+// GetSpec returns the Spec of this resource
+func (p *Person) GetSpec(_ runtime.Scheme) (genruntime.ARMTransformer, error) {
+    return &p.Spec, nil
+}
+
+// NewStatus returns a new blank status, ready to be populated
+func (p *Person) NewStatus(_ runtime.Scheme) (genruntime.FromARMConverter, error) {
+    return &Person_Status{}, nil
+}
+
+// SetStatus accepts a status and updates this resource
+// Only a status of the expected type is handled; anything else will error
+func (p *Person) SetStatus(status interface{}) error {
+    s, ok := status.(*Person_Status);
+    if !ok {
+        return errors.Errorf(
+            "expected Status of type %T, but received %T", p.Status, status)
+    } 
+
+    p.Status = *s
+    return nil
+}
+```
+
+### Implementation for storage versions
+
+For the hub Storage version of resources, the generated implementation will transparently handle conversion between versions as dictated by the `OriginalVersion()` function described earlier.
+
+We pivot to the original version by using the `OriginalGVK()` function to obtain a new, empty resource from the schema.
+
+``` go
+// GetSpec returns the spec of the current type using 
+// the original API version from when the CR was created
+func (p *Person) GetSpec(scheme runtime.Scheme) (genruntime.ARMTransformer, error) {
+    gvk := p.OriginalGVK()
+    
+    // Use gvk to create an empty resource of the expected version 
+    originalObj, err := scheme.New(gvk)
+    if err != nil {
+        return errors.Wrapf(
+            err, "unable to create requested version %s of resource", gvk)
+    }
+    resource, ok := originalObj.(genruntime.GeneratedResource)
+    if !ok {
+        return errors.Wrapf(
+            err, "unable to convert %T to genruntime.GeneratedResource", originalObj)
+    }
+
+    // Extract the Spec and populate
+    resultSpec := resource.GetSpec()
+    resultSpec.ConvertFromSpec(p.Spec)
+
+    return &resultSpec, nil
+}
+```
+
+The `NewStatus()` method needs to pivot to the original version of the resource so that can return the right kind of status.
+
+``` go
+func (p *Person) NewStatus(scheme runtime.Scheme) (genruntime.ARMTransformer, error) {
+    gvk := p.OriginalGVK()
+
+    // Use gvk to create an empty resource of the expected version
+    original, err := scheme.New(gvk)
+    if err != nil {
+        return errors.Wrapf(
+            err, "unable to create requested version %s of resource", gvk)
+    }
+    resource, ok := originalObj.(genruntime.GeneratedResource)
+    if !ok {
+        return errors.Wrapf(
+            err, "unable to convert %T to genruntime.GeneratedResource", originalObj)
+    }
+
+    // Return an empty Status from the empty resource
+    return resource.NewStatus(scheme)
+}
+```
+
+After the reconciler has populated an empty `Status` instance, `SetStatus()` is called for the update. It will be the appropriate API `Spec` as that is returned by `NewStatus()`, so all we need to do is to populate our Hub status.
```

Comment:
```suggestion
After the reconciler has populated an empty `Status` instance, `SetStatus()` is called for the update. It will be the appropriate API `Status` as that is returned by `NewStatus()`, so all we need to do is to populate our Hub status.
```

Reply:
I think?

Reply:
Correct. Fixed.